WearComplicationProvidersTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/IconProviderService.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.wear.wearcomplicationproviderstestsuite
+
+import android.content.ComponentName
+import android.graphics.drawable.Icon
+import android.support.wearable.complications.ComplicationData
+import android.support.wearable.complications.ComplicationManager
+
+/**
+ * A complication provider that supports only [ComplicationData.TYPE_ICON] and cycles through
+ * a few different icons on each tap.
```

Comment:
It's important you call out why you are using a custom version of this class. I would add an additional comment here and the other files... maybe something like this...

Note: Uses a custom version of ComplicationProviderService to support coroutines, so data operations (specifically, calls to DataStore) can be supported directly in the onComplication update. If you don't have any of these operations, you can use the normal approach.

Reply:
Added in the latest commit! I also made `NoDataProviderService` subclass `ComplicationProviderService` directly since it doesn't call any suspending functions, and linked to it from the other services.